### PR TITLE
added all of the required back_end work

### PR DESCRIPTION
**Pull request by JordanRoth**
There's a text file called high_scores.txt that has five rows which represent the five highest scores achieved. Every time a game over happens, it checks if the score should be added and then adds it and writes the file. When the game starts, it reads the file and initializes the high score system.

### CODE REVIEW

File: src/high_scores.hpp [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+	bool init();
+
+	// Retrieve the current top 5 scores
+	std::vector<int> get_high_scores();
+
+	// Insert the current score into the top 5 if applicable
+	void check_score_and_insert(int score);
+
+private:
+	std::string high_scores_file = "../src/high_scores.txt";
```
Comment (JordanRoth) [inline src/high_scores.hpp:22]: @EmceeEscher may not work on Mac :/

File: src/high_scores.txt [hunk 2]
```diff
@@ -0,0 +1,5 @@
+432
+331
+225
+209
+11
```
Comment (JordanRoth) [inline src/high_scores.txt:5]: This one is the result of my local testing... proof of concept :)

**Comment by EmceeEscher** [comment]
@theochit are you planning to add to this PR for the frontend, or should we try to merge it and create a separate one?

cc @JordanRoth

**Comment by theochit** [comment]
@EmceeEscher I'm working on this branch, hopefully I'll finish it today

**Review by theochit — APPROVED** [review]

**Comment by theochit** [comment]
I ended up going back to master to test something, and solving the problem there. I'll merge this branch and make a new one after all

**Pull request merged**